vip-api/src/main/java/fr/insalyon/creatis/vip/api/SpringSecurityConfig.java

```diff
@@ -72,41 +91,64 @@
     @Autowired
     private Environment env;
 
+    @Autowired
+    private ApiBusiness apiBusiness;
+
+
+    @Override
+    protected SessionAuthenticationStrategy sessionAuthenticationStrategy() {
+        return new RegisterSessionAuthenticationStrategy(new SessionRegistryImpl());
+    }
+
+    @Autowired
+    public void configureGlobal(AuthenticationManagerBuilder auth) {
+        KeycloakAuthenticationProvider keycloakAuthenticationProvider = keycloakAuthenticationProvider();
+        keycloakAuthenticationProvider.setGrantedAuthoritiesMapper(grantedAuthoritiesMapper());
+        auth.authenticationProvider(keycloakAuthenticationProvider);
+    }
+
     @Override
     protected void configure(HttpSecurity http) throws Exception {
+        super.configure(http);
         http
-            .authorizeRequests()
+                .authorizeRequests()
                 .antMatchers("/rest/platform").permitAll()
                 .antMatchers("/rest/authenticate").permitAll()
+                .antMatchers("/rest/register").permitAll() //signup a user to VIP
```

Comment:
This should be secured to only let shanoir access it.

Reply:
done